### PR TITLE
notation cert command updates

### PR DESCRIPTION
**Pull request by patrickzheng200**
1. Added confirmation to 'notation cert delete' command.
2. Updated 'notation cert generate-test' to generate self-signed certificate.
3. Removed tsa support for now, and added signingAuthority.

Signed-off-by: Patrick Zheng <patrickzheng@microsoft.com>

### CODE REVIEW

File: cmd/notation/cert.go [hunk 1]
```diff
@@ -419,6 +427,16 @@ func removeCertsCore(storeType, namedStore, cert string, all bool, errorSlice []
...
+			prompt := fmt.Sprintf("Are you sure you want to delete: %q?", path)
+			confirmed, err := ioutil.AskForConfirmation(os.Stdin, prompt, confirmed)
+			if err != nil {
+				errorSlice = append(errorSlice, fmt.Errorf("%s with error \"%s\"", path, err.Error()))
+				return errorSlice
+			}
+			if !confirmed {
+				return errorSlice
+			}
+
```
Comment (chloeyin) [inline cmd/notation/cert.go:439]: I suggest separate `removeCertsCore` into two functions, e.g.: `removeSomeCerts` and `removeAllCerts`

File: cmd/notation/cert.go [hunk 2]
```diff
@@ -312,7 +315,7 @@ func listCerts(opts *certListOpts) error {
 			}
 		}
 
-		paths = dir.Path.ConfigFS.ListAllPath(dir.TrustStoreDir, "x509", "tsa", namedStore)
+		paths = dir.Path.ConfigFS.ListAllPath(dir.TrustStoreDir, "x509", "signingAuthority", namedStore)
```
Comment (JeyJeyGao) [inline cmd/notation/cert.go:318]: Why?
According to spec,
```
Timestamping Certificates: The x509/tsa trust store contains named stores with Time Stamping Authority (TSA) root certificates. NOTE TSA based timestamping will not be available in notation RC1.
```

Comment (patrickzheng200) [reply cmd/notation/cert.go:318]: We will have 3 types, namely ca, tsa, and signingAuthority. We will support tsa after RC.1.

File: cmd/notation/cert.go (hunk 1 above)
Comment (patrickzheng200) [reply cmd/notation/cert.go:428]: > I suggest split`removeCertsCore` into two functions, e.g.: `removeSomeCerts` and `removeAllCerts`

Good idea. Renamed to removeAllCerts and removeCert.

File: cmd/notation/cert.go [hunk 3]
```diff
@@ -129,7 +131,7 @@ func certRemoveCommand(opts *certRemoveOpts) *cobra.Command {
 		opts = &certRemoveOpts{}
 	}
 	command := &cobra.Command{
-		Use:     "delete [-t type] -s name {--all | fileName}",
+		Use:     "delete -t <type> -s <name> [-y] (--all | fileName)",
```
Comment (binbin-li) [inline cmd/notation/cert.go:134]: should we use `<fileName>`?

File: cmd/notation/cert.go [hunk 4]
```diff
@@ -404,31 +402,54 @@ func removeCerts(opts *certRemoveOpts) error {
...
+
+// removeCert deletes a specific certificate file from the User level
+// trust store, namely truststore/x509/storeType/namedStore/cert
+func removeCert(storeType, namedStore, cert string, confirmed bool, errorSlice []error) []error {
+	path, err := dir.Path.UserConfigFS.GetPath(dir.TrustStoreDir, "x509", storeType, namedStore, cert)
+	if err == nil {
+		prompt := fmt.Sprintf("Are you sure you want to delete: %q?", path)
+		confirmed, err := ioutil.AskForConfirmation(os.Stdin, prompt, confirmed)
+		if err != nil {
 			errorSlice = append(errorSlice, fmt.Errorf("%s with error \"%s\"", path, err.Error()))
```
Comment (binbin-li) [inline cmd/notation/cert.go:437]: Maybe we can replace `\"%s\"` with `%q`

**Review by binbin-li — APPROVED** [review]
minor comments, lgtm

File: cmd/notation/cert.go (hunk 3 above)
Comment (patrickzheng200) [reply cmd/notation/cert.go:134]: Nice catch!

**Pull request merged**